src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1190,18 +1190,25 @@ void SystemMapDCmd::execute(DCmdSource source, TRAPS) {
   MemMapPrinter::print_all_mappings(output(), _human_readable.value());
 }
 
+static constexpr char default_filename[] = "vm_memory_map_<pid>.txt";
+
 SystemDumpMapDCmd::SystemDumpMapDCmd(outputStream* output, bool heap) :
     DCmdWithParser(output, heap),
   _human_readable("-H", "Human readable format", "BOOLEAN", false, "false"),
-  _filename("-F", "file path (defaults: \"vm_memory_map_<pid>.txt\")", "STRING", false) {
+  _filename("-F", "file path", "STRING", false, default_filename) {
   _dcmdparser.add_dcmd_option(&_human_readable);
   _dcmdparser.add_dcmd_option(&_filename);
 }
 
 void SystemDumpMapDCmd::execute(DCmdSource source, TRAPS) {
-  stringStream default_name;
-  default_name.print("vm_memory_map_%d.txt", os::current_process_id());
-  const char* name = _filename.is_set() ? _filename.value() : default_name.base();
+  stringStream defaultname;
+  const char* name = nullptr;
+  if (::strcmp(default_filename, _filename.value()) == 0) {
+    defaultname.print("vm_memory_map_%d.txt", os::current_process_id());
+    name = defaultname.base();
+  } else {
+    name = _filename.value();
+  }
```

Comment:
This code should be considering if a default it specified or not, in case the specified value is identical to what is contained in `default_filename`.  With the current solution, if the user literally specifies vm_memory_map_<pid>.txt, the <pid> part will be expanded to the actual pid. See how [JDK-8323546](https://bugs.openjdk.org/browse/JDK-8323546) handled this.

Reply:
Oops yes should definitely be using _filename.is_set() rather than strcmp.

Reply:
I filed [JDK-8334164](https://bugs.openjdk.org/browse/JDK-8334164) for not using `_filename.is_set()` and [JDK-8334145](https://bugs.openjdk.org/browse/JDK-8334145) for the issue with \<pid\> missing from the default filename in the jcmd.l.